### PR TITLE
DEPR: Remove df.reduction(level)

### PR DESCRIPTION
**Pull request by mroeschke**
Introduced in https://github.com/pandas-dev/pandas/pull/40869

### CODE REVIEW

**Comment by jbrockmendel** [comment]
test_frame_consistency

File: pandas/tests/groupby/test_allowlist.py [hunk 1]
```diff
@@ -86,19 +84,14 @@ def test_regression_allowlist_methods(raw_frame, op, level, axis, skipna, sort):
...
-        grouped = frame.groupby(level=level, axis=axis, sort=sort)
+        grouped = frame.groupby("first", axis=axis, sort=sort)
         result = getattr(grouped, op)()
-        expected = getattr(frame, op)(level=level, axis=axis)
-        if sort:
-            expected = expected.sort_index(axis=axis, level=level)
-        tm.assert_frame_equal(result, expected)
+    # Previously compared to frame.op(level=...), but level removed in 2.0
+    # TODO: Assert something better
+    assert isinstance(result, DataFrame)
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_allowlist.py:94]: I'd be good with a tracking issue for this. I'd guess this test is largely redundant, but we might not be testing the combination of all these arguments at the same time anywhere else.

**Review by rhshadrach — COMMENTED** [review]
lgtm, just a question on the old whatsnew.

File: doc/source/whatsnew/v0.15.2.rst [hunk 2]
```diff
@@ -154,11 +154,13 @@ Other enhancements:
...
-     :okwarning:
+  .. code-block:: python
 
-     s = pd.Series([False, True, False], index=[0, 0, 1])
-     s.any(level=0)
+     >>> s = pd.Series([False, True, False], index=[0, 0, 1])
+     >>> s.any(level=0)
+     0     True
+     1    False
+     dtype: bool
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply doc/source/whatsnew/v0.15.2.rst:163]: I don't have a preference either. I was mainly matching the other `.. code-block:: python`s in this `Other enhancements:` section for consistency.

I think if we fully tackle freezing whatsnews https://github.com/pandas-dev/pandas/issues/6856 I would prefer the ipython version

File: pandas/tests/groupby/test_allowlist.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/groupby/test_allowlist.py:94]: Agreed. I'll create an issue and add the reference here.

**Pull request merged**